[Bug] Client Crashing on copying BookBot books

### Describe the bug

copying bookbot written book will crash meteor

LOG:
Reported exception thrown!
net.minecraft.class_148: mouseClicked event handler
	at net.minecraft.class_437.method_25412(class_437.java:495)
	at net.minecraft.class_312.method_1601(class_312.java:94)
	at net.minecraft.class_312.method_22686(class_312.java:165)
	at net.minecraft.class_1255.execute(class_1255.java:101)
	at net.minecraft.class_312.method_22684(class_312.java:165)
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:36)
	at org.lwjgl.system.JNI.invokeV(Native Method)
	at org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3101)
	at com.mojang.blaze3d.systems.RenderSystem.flipFrame(RenderSystem.java:172)
	at net.minecraft.class_1041.method_15998(class_1041.java:310)
	at net.minecraft.class_310.method_1523(class_310.java:1137)
	at net.minecraft.class_310.method_1514(class_310.java:728)
	at net.minecraft.client.main.Main.main(Main.java:217)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:568)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: java.lang.OutOfMemoryError: Out of stack space.
	at org.lwjgl.system.MemoryStack.nmalloc(MemoryStack.java:314)
	at org.lwjgl.system.MemoryStack.nUTF8(MemoryStack.java:686)
	at org.lwjgl.glfw.GLFW.glfwSetClipboardString(GLFW.java:4385)
	at net.minecraft.class_3872.md7eec4e$lambda$onInit$0$1(class_3872.java:566)
	at net.minecraft.class_4185.method_25306(class_4185.java:40)
	at net.minecraft.class_4264.method_25348(class_4264.java:16)
	at net.minecraft.class_339.method_25402(class_339.java:121)
	at net.minecraft.class_4069.method_25402(class_4069.java:27)
	at net.minecraft.class_3872.method_25402(class_3872.java:294)
	at net.minecraft.class_312.method_1611(class_312.java:94)
	at net.minecraft.class_437.method_25412(class_437.java:490)
	... 19 more
13:24:50.599
meteordevelopment.meteorclient.MeteorClient
Thread-5
Saving
13:24:51.971
launcher
main

AL lib: (EE) alc_cleanup: 1 device not closed
13:24:51.974
monitor
Process Monitor
Process crashed with exit code -1

### Steps to reproduce

```bash
get a book and quill
run bookbot
open the written book
click copy
client will crash
```


### Meteor Version

0.4.6

### Minecraft Version

1.17.1

### Operating System

Windows

### Before submitting a bug report

- [X] This bug wasn't already reported.
(I have checked every bug report on github)

- [x] This is a valid bug.
(I am able to reproduce this on the latest dev build)

--- a/src/main/java/meteordevelopment/meteorclient/mixin/BookScreenMixin.java
+++ b/src/main/java/meteordevelopment/meteorclient/mixin/BookScreenMixin.java
@@ -8,6 +8,7 @@ package meteordevelopment.meteorclient.mixin;
 import it.unimi.dsi.fastutil.io.FastByteArrayOutputStream;
 import meteordevelopment.meteorclient.gui.GuiThemes;
 import meteordevelopment.meteorclient.gui.screens.EditBookTitleAndAuthorScreen;
+import meteordevelopment.meteorclient.utils.player.ChatUtils;
 import net.minecraft.client.gui.screen.Screen;
 import net.minecraft.client.gui.screen.ingame.BookScreen;
 import net.minecraft.client.gui.widget.ButtonWidget;
@@ -20,6 +21,8 @@ import net.minecraft.nbt.NbtString;
 import net.minecraft.text.Text;
 import net.minecraft.util.Hand;
 import org.lwjgl.glfw.GLFW;
+import org.lwjgl.system.MemoryStack;
+import org.lwjgl.system.MemoryUtil;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.Shadow;
 import org.spongepowered.asm.mixin.injection.At;
@@ -63,7 +66,17 @@ public class BookScreenMixin extends Screen {
                         e.printStackTrace();
                     }
 
-                    GLFW.glfwSetClipboardString(mc.getWindow().getHandle(), Base64.getEncoder().encodeToString(bytes.array));
+                    String encoded = Base64.getEncoder().encodeToString(bytes.array);
+                    
+                    @SuppressWarnings("resource")
+                    long available = MemoryStack.stackGet().getPointer();
+                    long size = MemoryUtil.memLengthUTF8(encoded, true);
+
+                    if (size > available) {
+                        ChatUtils.error("Could not copy to clipboard: Out of memory.");
+                    } else {
+                        GLFW.glfwSetClipboardString(mc.getWindow().getHandle(), encoded);
+                    }
                 })
                 .position(4, 4)
                 .size(120, 20)

